CMakeLists.txt

```diff
@@ -190,9 +190,13 @@ if(CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT)
 endif(CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT)
 
 if(HIOP_BUILD_STATIC)
-  add_library(hiop STATIC $<TARGET_OBJECTS:hiopOptimization>
+  add_library(hiop STATIC
+    $<TARGET_OBJECTS:hiopOptimization>
     $<TARGET_OBJECTS:hiopLinAlg>
     $<TARGET_OBJECTS:hiopUtils>)
+  if(HIOP_WITH_KRON_REDUCTION)
+    target_sources(hiop PUBLIC $<TARGET_OBJECTS:hiopKronRed>)
+  endif(HIOP_WITH_KRON_REDUCTION)
```

Comment:
Probably a good idea to make the same change for shared HiOp library build, as well.